gc/base/VirtualMemory.cpp

```diff
@@ -163,11 +163,14 @@ MM_VirtualMemory::reserveMemory(J9PortVmemParams* params)
 
 #if defined(OMR_GC_DOUBLE_MAP_ARRAYLETS)
 void*
-MM_VirtualMemory::doubleMapArraylet(MM_EnvironmentBase *env, void* arrayletLeaves[], UDATA arrayletLeafCount, UDATA arrayletLeafSize, UDATA byteAmount, struct J9PortVmemIdentifier *newIdentifier, UDATA pageSize)
+MM_VirtualMemory::doubleMapArraylet(MM_EnvironmentBase *env, void* arrayletLeaves[], UDATA arrayletLeafCount, UDATA arrayletLeafSize, UDATA byteAmount, struct J9PortVmemIdentifier *newIdentifier, UDATA pageSize, bool isLargePages)
 {
 	OMRPORT_ACCESS_FROM_OMRVM(_extensions->getOmrVM());
 	struct J9PortVmemIdentifier *oldIdentifier = &_identifier;
 	uintptr_t mode = OMRPORT_VMEM_MEMORY_MODE_READ | OMRPORT_VMEM_MEMORY_MODE_WRITE | OMRPORT_VMEM_MEMORY_MODE_COMMIT;
+	if (isLargePages) {
+		mode |= OMRPORT_VMEM_MEMORY_MODE_HUGE_PAGES;
```

Comment:
You set this flag in `omrvmem_reserve_memory_ex()` as well. Is it necessary to set it in both places? I would prefer to set it in one place only if possible.

Reply:
I see, this different path. I am taking this comment back.

Reply:
Here we need to set mode with `OMRPORT_VMEM_MEMORY_MODE_HUGE_PAGES` because the one at `omrvmem_reserve_memory_ex()` is for the heap and this method `doubleMapArraylet()` won't call `omrvmem_reserve_memory_ex()` but it will call `omrvmem_get_contiguous_region_memory()` to double map the arraylets.

Reply:
still, does omrvmem_get_contiguous_region_memory really require huge page flag? On heap creation yes, but later when double mapping is performed perhaps it can magically extract that info from heap's file descriptor.